x/poe/keeper/keeper.go

```diff
@@ -40,29 +42,43 @@ func NewKeeper(
 	}
 
 	return Keeper{
-		codec:       marshaler,
-		storeKey:    key,
-		paramStore:  paramSpace,
-		twasmKeeper: twasmK,
+		codec:             marshaler,
+		storeKey:          key,
+		paramStore:        paramSpace,
+		twasmKeeper:       twasmK,
+		contractAddrCache: make(map[types.PoEContractType]sdk.AccAddress),
 	}
 }
 
 // SetPoEContractAddress stores the contract address for the given type. If one exists already then it is overwritten.
 func (k Keeper) SetPoEContractAddress(ctx sdk.Context, ctype types.PoEContractType, contractAddr sdk.AccAddress) {
 	store := ctx.KVStore(k.storeKey)
 	store.Set(poeContractAddressKey(ctype), contractAddr.Bytes())
+	k.contractAddrCache[ctype] = contractAddr
```

Comment:
This function is only called during bootstrap or genesis import in production code. I think we are safe without a RW lock now but I am not sure for the future as this method is public. WDYT?